### DISCUSSION TITLE
Battery invert power

### DISCUSSION THREAD

**Discussion by vinceg0267** [slipx06/sunsynk-power-flow-card#210] [General]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Despite the Power and Current sensors of the battery provides a positive number for battery charge and negative number for battery discharge and invert_power set to true, during charging the card shows a positive current value and a negative power value, while the dot flows in the right direction. 

### Expected behaviour

During charging the dot must flow from the inverter to the battery and the power and current values must be positive. During the discharge phase the dot must move from the battery towards the inverter and the power and current values must be negative

### Card Version

4.9.1

### Home Assistant Version

OS 2023.12.4

**Comment by slipx06** [comment]
The card has been designed such that the battery power and current should show a negative value for battery charge and a positive value for battery discharge unless show_absolute has been set to true. You should start by looking at what your native sensor provides and then use invert_power to match what the card expects if needed.

EDIT
It seems that your sensor is providing a negative number for battery discharge and a positive number for battery charge.  So you will need to use invert_power. This will not invert the current so you will end up with your battery power and current being inverted. The work around is to set show_absolute: true or you can create a template sensor to invert your battery current so that they match i.e. negative power with negative current and vice versa
